subcommands/pkg/uninstall.go

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2025 Eric Faurot <eric.faurot@plakar.io>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+package pkg
+
+import (
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"slices"
+
+	"github.com/PlakarKorp/kloset/repository"
+	"github.com/PlakarKorp/plakar/appcontext"
+	"github.com/PlakarKorp/plakar/plugins"
+	"github.com/PlakarKorp/plakar/subcommands"
+	"github.com/PlakarKorp/plakar/utils"
+)
+
+func init() {
+	subcommands.Register(func() subcommands.Subcommand { return &PkgUninstall{} },
+		subcommands.BeforeRepositoryOpen,
+		"pkg", "uninstall")
+}
+
+type PkgUninstall struct {
+	subcommands.SubcommandBase
+	Args []string
+}
+
+func (cmd *PkgUninstall) Parse(ctx *appcontext.AppContext, args []string) error {
+	flags := flag.NewFlagSet("pkg uninstall", flag.ExitOnError)
+	flags.Usage = func() {
+		fmt.Fprintf(flags.Output(), "Usage: %s",
+			flags.Name())
+		fmt.Fprintf(flags.Output(), "\nOPTIONS:\n")
+		flag.PrintDefaults()
+	}
+
+	flags.Parse(args)
+
+	cmd.Args = flags.Args()
+
+	return nil
+}
+
+func (cmd *PkgUninstall) Execute(ctx *appcontext.AppContext, _ *repository.Repository) (int, error) {
+	cachedir, err := utils.GetCacheDir("plakar")
+	if err != nil {
+		return 1, err
+	}
+	cachedir = filepath.Join(cachedir, "plugins")
+
+	dataDir, err := utils.GetDataDir("plakar")
+	if err != nil {
+		return 1, err
+	}
+
+	pluginsDir := filepath.Join(dataDir, "plugins")
+	names, err := plugins.ListDir(ctx, pluginsDir)
+	if err != nil {
+		return 1, err
+	}
+
+	for _, name := range cmd.Args {
+		if !slices.Contains(names, name) {
+			return 1, fmt.Errorf("plugin %q is not installed", name)
+		}
+		err := os.Remove(filepath.Join(pluginsDir, name))
+		if err != nil {
+			return 1, fmt.Errorf("failed to remove %q: %w", name, err)
+		}
+		extlen := len(filepath.Ext(name))
+		pluginCache := filepath.Join(cachedir, name[:len(name)-extlen])
+		err = os.RemoveAll(pluginCache)
+		if err != nil {
+			return 1, fmt.Errorf("failed to remove cache for %q: %w", name, err)
+		}
```

Comment:
same here, i'd make this silent is the directory doesn't exist.

Reply:
agreed too